__tests__/cookie-banner.test.js

```diff
@@ -0,0 +1,168 @@
+/* eslint-env jest */
+
+const { setupPage } = require('../lib/jest-utilities.js')
+const configPaths = require('../config/paths.json')
+const PORT = configPaths.testPort
+
+let page
+let baseUrl = 'http://localhost:' + PORT
+
+beforeEach(async () => {
+  page = await setupPage()
+})
+
+afterEach(async () => {
+  await page.deleteCookie({ name: 'design_system_cookies_policy' })
+  await page.close()
+})
+
+const COOKIE_BANNER_SELECTOR = '[data-module="govuk-cookie-banner"]'
+
+describe('Cookie banner', () => {
+  describe('is hidden', () => {
+    it('when JavaScript is disabled', async () => {
+      await page.setJavaScriptEnabled(false)
+      await page.goto(`${baseUrl}`, { waitUntil: 'load' })
+      const isCookieBannerHidden = await page.waitForSelector(COOKIE_BANNER_SELECTOR, { visible: false })
+      expect(isCookieBannerHidden).toBeTruthy()
+    })
+
+    it('is hidden when JavaScript is enabled and the consent cookie version is valid', async () => {
+      await page.setCookie({ name: 'design_system_cookies_policy', value: '{"analytics":true, "version":1', domain: 'design-system.service.gov.uk' })
```

Comment:
I can't see the `domain` key in any of the expected values when we're checking cookies in other tests – is this needed as part of the object?

Reply:
Thanks, I've amended this. The tests complain if I don't pass in at least `url` or `domain`. I've changed it so we now consistently pass in `url: baseUrl` rather than hardcoding a domain within the tests.